src/Lib.P2P/PubSub/FloodRouter.cs

```diff
@@ -0,0 +1,307 @@
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Lib.P2P.Protocols;
+using ProtoBuf;
+using Semver;
+
+namespace Lib.P2P.PubSub
+{
+    /// <summary>
+    ///   The original flood sub router.
+    /// </summary>
+    public class FloodRouter : IPeerProtocol, IMessageRouter
+    {
+        private static ILog log = LogManager.GetLogger(typeof(FloodRouter));
+
+        private MessageTracker tracker = new MessageTracker();
+        private ConcurrentDictionary<string, string> localTopics = new ConcurrentDictionary<string, string>();
+
+        /// <summary>
+        ///   The topics of interest of other peers.
+        /// </summary>
+        public TopicManager RemoteTopics { get; set; } = new TopicManager();
+
+        /// <inheritdoc />
+        public event EventHandler<PublishedMessage> MessageReceived;
+
+        /// <inheritdoc />
+        public string Name { get; } = "floodsub";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1, 0);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <summary>
+        ///   Provides access to other peers.
+        /// </summary>
+        public SwarmService SwarmService { get; set; }
+
+        /// <inheritdoc />
+        public Task StartAsync()
+        {
+            log.Debug("Starting");
+
+            SwarmService.AddProtocol(this);
+            SwarmService.ConnectionEstablished += Swarm_ConnectionEstablished;
+            SwarmService.PeerDisconnected += Swarm_PeerDisconnected;
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public Task StopAsync()
+        {
+            log.Debug("Stopping");
+
+            SwarmService.ConnectionEstablished -= Swarm_ConnectionEstablished;
+            SwarmService.PeerDisconnected -= Swarm_PeerDisconnected;
+            SwarmService.RemoveProtocol(this);
+            RemoteTopics.Clear();
+            localTopics.Clear();
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            while (true)
+            {
+                var request = await ProtoBufHelper.ReadMessageAsync<PubSubMessage>(stream, cancel)
+                   .ConfigureAwait(false);
+                ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)